### PR TITLE
Add synthetic video generation to "bandwidth" sample, and measure rampup

### PR DESCRIPTION
**Pull request by alvestrand**
**Description**
This allows checking how long it takes for resolution to adapt after
starting the call.

**Purpose**
Verifying that rampup takes too long.

### CODE REVIEW

File: src/content/peerconnection/bandwidth/js/main.js [hunk 1]
```diff
@@ -288,3 +314,43 @@ window.setInterval(() => {
...
+  const ctx = canvas.getContext('2d');
+  const stream = canvas.captureStream();
+
+  let count = 0;
+  setInterval(() => {
+    // Use relatively-prime multipliers to get a color roll
+    const r = Math.abs((count*2)%512 - 256);
+    const g = Math.abs((count*3)%512 - 256);
+    const b = Math.abs((count*5)%512 - 256);
+    ctx.fillStyle = `rgb(${r}, ${g}, ${b})`;
```
Comment (dogben) [inline src/content/peerconnection/bandwidth/js/main.js:338]: rgb accepts numbers between 0 and 255 (incl) or percentages from 0% to 100%.

File: src/content/peerconnection/bandwidth/js/main.js [hunk 2]
```diff
@@ -288,3 +314,43 @@ window.setInterval(() => {
...
+    count += 1;
+    const boxSize=80;
+    ctx.fillRect(0, 0, width, height);
+    // Add some bouncing boxes in contrast color to add a little more noise.
+    const rContrast = (r + 128)%255;
+    const gContrast = (g + 128)%255;
+    const bContrast = (b + 128)%255;
```
Comment (dogben) [inline src/content/peerconnection/bandwidth/js/main.js]: Did you perhaps mean 256 rather than 255 here?

**Review by dogben — APPROVED** [review]

File: src/content/peerconnection/bandwidth/js/main.js (hunk 2 above)
Comment (alvestrand) [reply src/content/peerconnection/bandwidth/js/main.js]: I woke up in the night and said to myself "%255 will only go to 254!" :-)
This was code I copied from RTCPeerConnection-helper.js in WPT; it was improper from before.

(Once this is landed, I intend to take the signal generator back into RTCPeerConnection-helper)

File: src/content/peerconnection/bandwidth/js/main.js (hunk 1 above)
Comment (alvestrand) [reply src/content/peerconnection/bandwidth/js/main.js:338]: The purpose of the fancy math is to create a signal that changes enough to give the codec something to do, but without to many bandwidth spikes. I may iterate once more over this.

Comment (dogben) [reply src/content/peerconnection/bandwidth/js/main.js:338]: Ok, so is it intentional to pass negative numbers to rgb? I assume they would be coerced to zero, but I'm not sure.

Comment (dogben) [reply src/content/peerconnection/bandwidth/js/main.js:338]: Oops, somehow I glossed over the Math.abs.

File: src/content/peerconnection/bandwidth/js/main.js [hunk 3]
```diff
@@ -315,6 +315,13 @@ window.setInterval(() => {
   });
 }, 1000);
 
+// Return a number between 0 and maxValue based on the input number,
+// so that the output changes smoothly up and down.
+function triangle(number, maxValue) {
+  const modulus = (maxValue + 1) * 2;
+  return Math.abs(number % modulus - maxValue);
```
Comment (dogben) [inline src/content/peerconnection/bandwidth/js/main.js:322]: triangle(511, 255) returns 256.

**Review by dogben — APPROVED** [review]

**Pull request merged**